internal/wrappers/mock/results-mock.go

```diff
@@ -42,11 +42,17 @@ func (r ResultsMockWrapper) GetAllResultsPackageByScanID(params map[string]strin
 	return &scaPackages, nil, nil
 }
 
-func (r ResultsMockWrapper) GetAllResultsByScanID(_ map[string]string) (
+func (r ResultsMockWrapper) GetAllResultsByScanID(params map[string]string) (
 	*wrappers.ScanResultsCollection,
 	*wrappers.WebError,
 	error,
 ) {
+	if params["scan-id"] == "MOCK_NO_VULNERABILITIES" {
+		return &wrappers.ScanResultsCollection{
+			TotalCount: 0,
+			Results:    nil,
```

Comment:
if results is not il but vulnerabilities is nil? can it override the empty list you did? can you have a test for it?

Reply:
We are creteing the Vulnerabilities from the Results. When there are no result, int the old behaviour the vulnerabilities were not initialized and were null. In the new behaviour they will be empty slice. 
I added a test that will check if the vulnerabilities we created are empty slice and not nill